drop-transfer/src/service.rs

```diff
@@ -114,85 +120,16 @@ impl Service {
         &self.state.storage
     }
 
-    fn trigger_peer_incoming(&mut self, xfer: Arc<crate::IncomingTransfer>) {
-        debug!(self.logger, "trigger_peer_incoming() called: {:?}", xfer);
-
-        let id = xfer.id();
-
-        if let Some(handle) = self.tasks.get(&id) {
-            if !handle.is_finished() {
-                return;
+    pub fn network_refresh(&mut self) {
+        match self.refresh_trigger.send(()) {
+            Ok(_) => {
+                trace!(self.logger, "Refresh trigger sent");
             }
-        }
-
-        let handle = check::spawn(
-            self.state.clone(),
-            xfer,
-            self.logger.clone(),
-            self.waiter.guard(),
-            self.stop.clone(),
-        );
-
-        self.tasks.insert(id, handle);
-    }
-
-    fn trigger_peer_outgoing(&mut self, xfer: Arc<crate::OutgoingTransfer>) {
-        debug!(self.logger, "trigger_peer_outgoing() called: {:?}", xfer);
-
-        let id = xfer.id();
-        if let Some(handle) = self.tasks.get(&id) {
-            if !handle.is_finished() {
-                return;
-            }
-        }
-
-        let handle = ws::client::spawn(
-            self.state.clone(),
-            xfer,
-            self.logger.clone(),
-            self.waiter.guard(),
-            self.stop.clone(),
-        );
-
-        self.tasks.insert(id, handle);
-    }
-
-    pub async fn set_peer_state(&mut self, addr: IpAddr, is_online: bool) {
-        {
-            if is_online {
-                let outgoing_transfers_to_trigger = {
-                    let xfers = self.state.transfer_manager.outgoing.lock().await;
-
-                    xfers
-                        .values()
-                        .filter(|state| {
-                            let peer = state.xfer.peer();
-                            peer == addr
-                        })
-                        .map(|state| state.xfer.clone())
-                        .collect::<Vec<_>>()
-                };
-
-                for xfer in outgoing_transfers_to_trigger {
-                    self.trigger_peer_outgoing(xfer);
-                }
-
-                let incoming_transfers_to_trigger = {
-                    let xfers = self.state.transfer_manager.incoming.lock().await;
-
-                    xfers
-                        .values()
-                        .filter(|state| {
-                            let peer = state.xfer.peer();
-                            peer == addr
-                        })
-                        .map(|state| state.xfer.clone())
-                        .collect::<Vec<_>>()
-                };
-
-                for xfer in incoming_transfers_to_trigger {
-                    self.trigger_peer_incoming(xfer);
-                }
+            Err(err) => {
+                error!(
+                    self.logger,
+                    "Could not send a refresh trigger(probably channel closed): {:?}", err
+                )
```

Comment:
I think you can simply ignore the error here:
```rust
let _ = self.refresh_trigger.send(());
```
The error happens when there are no receivers alive. Which is not an error in our case